openfl-tutorials/interactive_api/PyTorch_MedMNIST_2D/envoy/medmnist_shard_descriptor.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""MedMNIST Shard Descriptor."""
+
+import logging
+import os
+from typing import List
+from medmnist.info import INFO, HOMEPAGE
+
+import numpy as np
+import requests
+
+from openfl.interface.interactive_api.shard_descriptor import ShardDataset
+from openfl.interface.interactive_api.shard_descriptor import ShardDescriptor
+
+logger = logging.getLogger(__name__)
+
+
+class MedMNISTShardDataset(ShardDataset):
+    """MedMNIST Shard dataset class."""
+
+    def __init__(self, x, y, data_type, rank=1, worldsize=1):
```

Comment:
Please add type annotations for all functions in this file. For eg: def __init__(self, x, y, data_type, rank: int = 1, worldsize: int = 1) -> None